File: Data/Models/Filme.cs
namespace TrabalhoAPI.Data.Models
{
    public class Filme
    {
        public int Id { get; set; }
        public string? Titulo { get; set; }
        public string? Sinopse { get; set; }
        public int Ano { get; set; }
        public int DiretorId { get; set; }
        public Diretor? Diretor { get; set; }
        public List<CatalogoFilme>? CatalogoFilmes { get; set; }
        public List<int>? CatalogoIds { get; set; }
        public List<Comentario>? Comentarios { get; set; }
        public List<GeneroFilme>? GeneroFilmes { get; set; }
        
    }
}

File: src/app/Filme.ts
import { CatalogoFilme } from "./CatalogoFilme";
import { Comentario } from "./Comentario";
import { Diretor } from "./Diretor";
import { GeneroFilmes } from "./GeneroFilmes";

export class Filme{
    id: number = 0;
    titulo: string = '';
    sinopse: string = '';
    ano: number = 0;
    diretorId: number = 0;
    diretor: Diretor | null = null;
    catalogoFilmes: CatalogoFilme[] =[];
    catalogoIds: number[] = [];
    comentarios: Comentario[] = [];
    generoFilmes: GeneroFilmes[] = [];
}